File: Exercise/Medium/12. Tokenize String into Substrings/Code/Tokenize String into Substrings.java
import java.util.Scanner;

public class TokenizeString {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        char delimiter = scanner.next().charAt(0);

        String[] tokens = input.split(String.valueOf(delimiter));

        for (String token : tokens) {
            System.out.println(token);
        }

        scanner.close();
    }
}

File: Exercise/Medium/12. Tokenize String into Substrings/Code/Tokenize String into Substrings.py
def tokenize_string(input_str, delimiter):
    tokens = []
    token = ""
    for ch in input_str:
        if ch == delimiter:
            if token:
                tokens.append(token)
                token = ""
        else:
            token += ch
    if token:
        tokens.append(token)
    return tokens

def main():
    input_str = input("Enter a string: ")
    delimiter = input("Enter a delimiter: ")[0]  

    tokens = tokenize_string(input_str, delimiter)

    for token in tokens:
        print(token)

if __name__ == "__main__":
    main()